overlord/patch/patch6_3.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package patch
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/snapcore/snapd/overlord/state"
+)
+
+func normChan(in string) string {
+	return strings.Join(strings.FieldsFunc(in, func(r rune) bool { return r == '/' }), "/")
+}
+
+// patch6_3:
+//  - ensure channel spec is valid
+func patch6_3(st *state.State) error {
```

Comment:
Nitpick, I haven't been in this part of the code lately but I have slight preference for:
```suggestion
func patch6dot3(st *state.State) error {
```

Reply:
Yeah, it looks a bit out of place but its consistent with the other patch6_{1,2} funcs.